CRM/Case/Form/CustomData.php

```diff
@@ -201,4 +198,44 @@ public function formatCustomDataChangesForDetail(array $params): string {
     return implode('<br/>', $formattedDetails);
   }
 
+  private function formatDisplayValue(mixed $value, int $customFieldId, string $customFieldDataType): string {
+    switch ($customFieldDataType) {
+      case 'Money':
+        // Money is special for non-US locales because at this point it's in human format so we don't try
+        // want to try to convert it.
+        return $value;
+
+      case 'File':
+        // File is tricky - updating the case file reuses the same File ID, which means saving urls is meaningless
+        // so every URL will just render the latest version of the file...
+        //
+        // For now we just render the URI (= the filename on the server)
+        // Q1: is leaking the hash suffix of the filename on server a data leak?
+        // Q2: should we update the handling so new files are saved with new IDs, and the old files are preserved
+        // and viewable in the data changed activities? or would the expectation be for old versions of files
+        // to be deleted
+
+        if (!empty($value['name'])) {
+          $uri = array_pop(explode(DIRECTORY_SEPARATOR, $value['name']));
```

Comment:
```suggestion
          $uri = CRM_Utils_File::cleanFileName(basename($value['name']));
```

basename will work on windows then too.